<?xml version="1.0" encoding="utf-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from urdf/toplevel.xacro            | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="yefan" xmlns:xacro="http://ros.org/wiki/xacro">
  <joint name="joint1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.346"/>
    <parent link="base_link"/>
    <child link="link2"/>
    <axis xyz="0 0 1"/>
    <limit effort="3000" lower="-3.14" upper="3.14" velocity="2.26893"/>
    <!--  bound +-185 limit effort ? and dynamics? -->
    <dynamics damping="125" friction="155"/>
  </joint>
  <joint name="joint2" type="revolute">
    <origin rpy="0 0 0" xyz="0.312 0 0.324"/>
    <parent link="link2"/>
    <child link="link3"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000" lower="-1.0472" upper="1.32645" velocity="2.00713"/>
    <!--  limit effort, boudns? range=136 ? ? and dynamics? -->
    <dynamics damping="72" friction="50"/>
  </joint>
  <joint name="joint3" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 1.075"/>
    <parent link="link3"/>
    <child link="link4"/>
    <axis xyz="0 -1 0"/>
    <limit effort="1000" lower="-1.83259" upper="1.8726" velocity="2.18166"/>
    <!--  limit effort, boudns? range=312 ? and dynamics? -->
    <dynamics damping="15" friction="48"/>
  </joint>
  <joint name="joint4" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.225"/>
    <parent link="link4"/>
    <child link="link5"/>
    <axis xyz="-1 0 0"/>
    <limit effort="940" lower="-3.14" upper="3.14" velocity="3.14159"/>
    <!--  limit effort =moment ? and dynamics? -->
    <dynamics damping="1" friction="1"/>
  </joint>
  <joint name="joint5" type="revolute">
    <origin rpy="0 0 0" xyz="1.28 0 0"/>
    <parent link="link5"/>
    <child link="link6"/>
    <axis xyz="0 -1 0"/>
    <limit effort="940" lower="-2.18166" upper="2.18166" velocity="3.14159"/>
    <!--  limit, bound? range=250 effort ? and dynamics? -->
    <dynamics damping="1" friction="1"/>
  </joint>
  <joint name="joint6" type="revolute">
    <origin rpy="0 0 0" xyz="0.215 0 0"/>
    <parent link="link6"/>
    <child link="link7"/>
    <axis xyz="-1 0 0"/>
    <limit effort="490" lower="-3.14" upper="3.14" velocity="4.53786"/>
    <!--  limit effort ? and dynamics? -->
    <dynamics damping="1" friction="1"/>
  </joint>
  <link name="base_link">
    <inertial>
      <mass value="550"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://fanuc/meshes/J1.STL"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://fanuc/meshes/J1.STL"/>
      </geometry>
    </visual>
  </link>
  <link name="link2">
    <inertial>
      <mass value="250"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 -0.346"/>
      <geometry>
        <mesh filename="package://fanuc/meshes/J2Base.STL"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.346"/>
      <geometry>
        <mesh filename="package://fanuc/meshes/J2Base.STL"/>
      </geometry>
    </visual>
  </link>
  <link name="link3">
    <inertial>
      <mass value="60"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="-0.312 0 -0.67"/>
      <geometry>
        <mesh filename="package://fanuc/meshes/J2Arm.STL"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="-0.312 0 -0.67"/>
      <geometry>
        <mesh filename="package://fanuc/meshes/J2Arm.STL"/>
      </geometry>
    </visual>
  </link>
  <link name="link4">
    <inertial>
      <mass value="30"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="-0.312 0 -1.745"/>
      <geometry>
        <mesh filename="package://fanuc/meshes/J3.STL"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="-0.312 0 -1.745"/>
      <geometry>
        <mesh filename="package://fanuc/meshes/J3.STL"/>
      </geometry>
    </visual>
  </link>
  <link name="link5">
    <inertial>
      <mass value="15"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="-0.312 0 -1.97"/>
      <geometry>
        <mesh filename="package://fanuc/meshes/J4.STL"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="-0.312 0 -1.97"/>
      <geometry>
        <mesh filename="package://fanuc/meshes/J4.STL"/>
      </geometry>
    </visual>
  </link>
  <link name="link6">
    <inertial>
      <mass value="20"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="-1.592 0 -1.97"/>
      <geometry>
        <mesh filename="package://fanuc/meshes/J5.STL"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="-1.592 0 -1.97"/>
      <geometry>
        <mesh filename="package://fanuc/meshes/J5.STL"/>
      </geometry>
    </visual>
  </link>
  <link name="link7">
    <inertial>
      <mass value="2"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="-1.807 0 -1.97"/>
      <geometry>
        <mesh filename="package://fanuc/meshes/J6.STL"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="-1.807 0 -1.97"/>
      <geometry>
        <mesh filename="package://fanuc/meshes/J6.STL"/>
      </geometry>
    </visual>
  </link>
  <gazebo reference="base_link">
    <material>Gazebo/Black</material>
  </gazebo>
  <gazebo reference="link2">
    <material>Gazebo/Yellow</material>
  </gazebo>
  <gazebo reference="link3">
    <material>Gazebo/Yellow</material>
  </gazebo>
  <gazebo reference="link4">
    <material>Gazebo/Yellow</material>
  </gazebo>
  <gazebo reference="link5">
    <material>Gazebo/Yellow</material>
  </gazebo>
  <gazebo reference="link6">
    <material>Gazebo/Yellow</material>
  </gazebo>
  <gazebo reference="link7">
    <material>Gazebo/White</material>
  </gazebo>
  <transmission name="joint1_transmssion">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint1">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="joint1_motor">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="joint1">
    <implicitSpringDamper>1</implicitSpringDamper>
  </gazebo>
  <transmission name="joint2_transmssion">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint2">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="joint2_motor">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="joint2">
    <implicitSpringDamper>1</implicitSpringDamper>
  </gazebo>
  <transmission name="joint3_transmssion">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint3">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="joint3_motor">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="joint3">
    <implicitSpringDamper>1</implicitSpringDamper>
  </gazebo>
  <transmission name="joint4_transmssion">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint4">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="joint4_motor">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="joint4">
    <implicitSpringDamper>1</implicitSpringDamper>
  </gazebo>
  <transmission name="joint5_transmssion">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint5">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="joint5_motor">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="joint5">
    <implicitSpringDamper>1</implicitSpringDamper>
  </gazebo>
  <transmission name="joint6_transmssion">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint6">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="joint6_motor">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="joint6">
    <implicitSpringDamper>1</implicitSpringDamper>
  </gazebo>
  <!-- <xacro:include filename="$(find fanuc)/urdf/additions.xacro"/> -->
</robot>

